src/Native/Runtime/ObjectLayout.cpp

```diff
@@ -38,9 +38,9 @@ void* Array::GetArrayData()
 }
 
 #ifndef DACCESS_COMPILE
-void Array::InitArrayLength(UInt32 length)
+void Array::InitArrayLength(UInt32 length, bool maybeNonZero)
 {
-    ASSERT(NULL == m_Length);
```

Comment:
This assert was hitting because the field for `m_Length` was no longer zero'd

Reply:
I think it would be better to delete the assert instead of the extra argument.